framework/web/Request.php

```diff
@@ -749,13 +749,20 @@ public function getHostInfo()
                 $this->_hostInfo = $http . '://' . trim(explode(',', $this->headers->get('X-Forwarded-Host'))[0]);
             } elseif ($this->headers->has('X-Original-Host')) {
                 $this->_hostInfo = $http . '://' . trim(explode(',', $this->headers->get('X-Original-Host'))[0]);
-            } elseif ($this->headers->has('Host')) {
-                $this->_hostInfo = $http . '://' . $this->headers->get('Host');
-            } elseif (isset($_SERVER['SERVER_NAME'])) {
-                $this->_hostInfo = $http . '://' . $_SERVER['SERVER_NAME'];
-                $port = $secure ? $this->getSecurePort() : $this->getPort();
-                if (($port !== 80 && !$secure) || ($port !== 443 && $secure)) {
-                    $this->_hostInfo .= ':' . $port;
+            } else {
+                if ($this->headers->has('Host')) {
+                    $this->_hostInfo = $http . '://' . $this->headers->get('Host');
+                } elseif (filter_has_var(INPUT_SERVER, 'SERVER_NAME')) {
+                    $this->_hostInfo = $http . '://' . filter_input(INPUT_SERVER, 'SERVER_NAME');
+                } elseif (isset($_SERVER['SERVER_NAME'])) {
+                    $this->_hostInfo = $http . '://' . $_SERVER['SERVER_NAME'];
+                }
+
+                if ($this->_hostInfo !== null) {
+                    $port = $secure ? $this->getSecurePort() : $this->getPort();
+                    if (($port !== 80 && !$secure) || ($port !== 443 && $secure)) {
+                        $this->_hostInfo .= ':' . $port;
+                    }
```

Comment:
I just realized the `Host` header can actually contain the port, if it’s not 80 or 443. It’s still probably good to ensure it gets added, but this code should check to make sure `$this->_hostInfo` doesn’t already specify a port before adding one.

Reply:
@brandonkelly, You mean something like this

Reply:
Just in case the `Host` port is not what we thought it would be, it’s probably safer to check if `$this->_hostInfo` ends in a port from the initial condition:

```php
if ($this->_hostInfo !== null && !preg_match('/:\d+$/', $this->_hostInfo)) {
    $port = $secure ? $this->getSecurePort() : $this->getPort();
    if (($port !== 80 && !$secure) || ($port !== 443 && $secure)) {
        $this->_hostInfo .= ':' . $port;
    }
```

Reply:
@brandonkelly, check last commit, pls

Reply:
Looks good!